source/common/secret/secret_impl.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include "envoy/api/v2/auth/cert.pb.h"
+#include "envoy/secret/secret.h"
+
+namespace Envoy {
+namespace Secret {
+
+class SecretImpl : public Secret {
```

Comment:
This should really be `class TlsCertificateConfigImpl : public Secret` or something like that, otherwise it won't work scale for other secret types.

Perhaps move it to `source/common/ssl` as well?

Reply:
Done